proxy/common/flags.py

```diff
@@ -523,6 +530,13 @@ def load_plugins(plugins: bytes) -> Dict[bytes, List[type]]:
                 # HttpWebServerRouteHandler route decorator adds a special
                 # staticmethod to return decorated function name
                 klass.__name__ if klass.__name__ != 'HttpWebServerRouteHandler' else klass.name())
+        # if a plugin need the HttpWebServerPlugin to be loaded, and if it's missing, we force the load of this plugin
```

Comment:
I am also not sure if we must add this logic inside `load_plugins` which is actually agnostic of any logic imposed by `proxy.py`.  All it knows is, load a list of plugins passed to it.  So adding a business logic like `enable plugin if / else` here won't be a cleaner solution.  Let's enforce HttpWebServerPlugin at a high level.

I was thinking of adding necessary checks within Flags module, but before `load_plugin` gets called.  Idea is that we should be able to detect before calling load_plugins, whether web server plugin should be loaded or not.

Reply:
> I was thinking of adding necessary checks within Flags module, but before load_plugin gets called. Idea is that we should be able to detect before calling load_plugins, whether web server plugin should be loaded or not.

I agree with you. My first idea was something like that, but my main problem was to get the "main" class of each plugin.  It makes us able to know if we must add the "--enable-web-server"/"HttpWebServerPlugin" or not. 
But, to do that, it seems that we are obliged to do all the process  as you currently do with
```
klass = getattr(
                importlib.import_module(
                    module_name.replace(
                        os.path.sep, text_(DOT))),
                klass_name)
```
to get the main class for each plugin.

If we check before the call of load_plugins, the behavior of this check will be more all less the same because, as I said before, it looks like we need to get the `klass` to know the main class of the plugin.